### PR TITLE
Web console: expose handoff API

### PR DESCRIPTION
**Pull request by vogievetsky**
Expose handoff API added in https://github.com/apache/druid/pull/16310

image

image

### CODE REVIEW

**Comment by johnImply** [comment]
Where on the UI are the Task Group IDs found?

If this is the "Group ID" value from the Tasks tab, then would it make more sense to implement the "Early Handoff" option on the Tasks tab instead of the Supervisor tab?

I understand it is the Supervisor that needs the direct instruction and not the Task ... but from a user perspective it is the task that I am cycling, not the Supervisor.

Thanks.  John

**Comment by georgew5656** [comment]
> Where on the UI are the Task Group IDs found?
> 
> If this is the "Group ID" value from the Tasks tab, then would it make more sense to implement the "Early Handoff" option on the Tasks tab instead of the Supervisor tab?
> 
> I understand it is the Supervisor that needs the direct instruction and not the Task ... but from a user perspective it is the task that I am cycling, not the Supervisor.
> 
> Thanks. John

the task's taskGroupId is under the streaming task -> ioConfig -> taskGroupId. i think it still makes sense to have it on the supervisor (since you can specify multiple tasks to handoff, and like you said the api is operating on the supervisor). it would be nice to have a more explicit group -> task_id mapping available somewhere in the superivsor though.

i have observed that the Statistics tab for the supervisor already shows tasks by ascending group id (1,2,3,4,5, etc) so maybe it could be there, or alternatively just in the status page

File: web-console/src/dialogs/supervisor-handoff-dialog/supervisor-handoff-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      action={async () => {
+        const resp = await Api.instance.post(
+          `/druid/indexer/v1/supervisor/${Api.encodePath(supervisorId)}/taskGroups/handoff`,
+          { taskGroupIds: groupIds.map(x => parseInt(x, 10)) },
+        );
+        return resp.data;
+      }}
+      confirmButtonText="Handoff supervisor"
+      successText="Supervisor handoff has been inititated"
+      failText="Could not initiate handoff for supervisor"
```
Comment (ektravel) [inline web-console/src/dialogs/supervisor-handoff-dialog/supervisor-handoff-dialog.tsx:47]:
```suggestion
      failText="Could not initiate handoff for the supervisor"
```

File: web-console/src/dialogs/supervisor-handoff-dialog/supervisor-handoff-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    <AsyncActionDialog
+      action={async () => {
+        const resp = await Api.instance.post(
+          `/druid/indexer/v1/supervisor/${Api.encodePath(supervisorId)}/taskGroups/handoff`,
+          { taskGroupIds: groupIds.map(x => parseInt(x, 10)) },
+        );
+        return resp.data;
+      }}
+      confirmButtonText="Handoff supervisor"
+      successText="Supervisor handoff has been inititated"
```
Comment (ektravel) [inline web-console/src/dialogs/supervisor-handoff-dialog/supervisor-handoff-dialog.tsx:46]:
```suggestion
      successText="Supervisor handoff has been initiated"
```

File: web-console/src/dialogs/supervisor-handoff-dialog/supervisor-handoff-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      failText="Could not initiate handoff for supervisor"
+      intent={Intent.PRIMARY}
+      onClose={onClose}
+      onSuccess={onSuccess}
+    >
+      <p>
+        Are you sure you want initiate handoff for the supervisor <Tag minimal>{supervisorId}</Tag>?
```
Comment (ektravel) [inline web-console/src/dialogs/supervisor-handoff-dialog/supervisor-handoff-dialog.tsx:53]:
```suggestion
        Are you sure you want to initiate handoff for the supervisor <Tag minimal>{supervisorId}</Tag>?
```

**Review by ektravel — COMMENTED** [review]
Left a few suggestions.

**Comment by johnImply** [comment]
Verbiage ... do we need to make a distinction between "supervisor handoff" and "task group handoff"? 

I thought a "supervisor handoff" involved cycling all tasks for the supervisor ... this is what we can do by "Open in Data Loader" and then Submit without making any changes to the supervisor spec.   

If there is any ambiguity here it might be safer to always reference "task group(s)" as the thing being handed off.    My understanding is that Supervisor is merely the orchestrator here.

**Review by ektravel — APPROVED** [review]
The changes look good.

**Pull request merged**